source/common/runtime/runtime_impl.cc

```diff
@@ -22,20 +26,19 @@ namespace Runtime {
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
 std::string RandomGeneratorImpl::uuid() {
-  int fd = open("/proc/sys/kernel/random/uuid", O_RDONLY);
-  if (-1 == fd) {
-    throw EnvoyException(fmt::format("unable to open uuid, errno: {}", strerror(errno)));
-  }
-
   char generated_uuid[UUID_LENGTH + 1];
-  ssize_t bytes_read = read(fd, generated_uuid, UUID_LENGTH);
-  close(fd);
-  generated_uuid[UUID_LENGTH] = '\0';
 
-  if (bytes_read != UUID_LENGTH) {
-    throw EnvoyException(fmt::format("cannot read the uuid: bytes read - {}, bytes expected - {}",
-                                     bytes_read, UUID_LENGTH));
-  }
+#if defined(__linux__) || defined(__APPLE__)
+  uuid_t uuid;
+  uuid_generate_random(uuid);
+  uuid_unparse(uuid, generated_uuid);
+#else
+  sprintf(generated_uuid, "%08lx-%04x-%04x-%04x-%04x%08lx",
+          static_cast<unsigned long>(arc4random()), static_cast<unsigned>(arc4random() & 0xffff),
```

Comment:
Probably best to generate the minimum number of random numbers required here before the `sprintf`. Also, would prefer `snprintf` if using C formatters for safety.